server/auth/types/authentication_type.ts

```diff
@@ -153,6 +151,11 @@ export abstract class AuthenticationType implements IAuthenticationType {
           return toolkit.notHandled();
         }
 
+        // allow optional authentication
+        if (this.authOptional(request)) {
```

Comment:
Why is this only in the block if there's no cookie or no valid cookie? I'm trying to determine if this block would need to be placed elsewhere as well. Is this API only called on initial login?

Reply:
Adding this code in this scope is necessary because if a user calls this API without logging in, it will throw a `No Authorization` exception. This API is not only called during login. The purpose of this PR is to ensure that when plugins call `core.capabilities.registerSwitcher` after logging in, the request can include authinfo.

Reply:
So, the code only needs to be added in this scope (when the user is not logged in: no cookie). In other cases, the user will already be logged in.

Reply:
What happens when the user name is not present?

Reply:
When user try to login, the following error will occur:
<img width="2309" alt="image" src="https://github.com/opensearch-project/security-dashboards-plugin/assets/71875999/387ee30b-ba99-4b0b-a5b3-8aa820eae2ca">